pkg/util/filesystem/util_unix.go

```diff
@@ -37,6 +37,10 @@ func IsUnixDomainSocket(filePath string) (bool, error) {
 	return true, nil
 }
 
+var Chmod = os.Chmod
+
+var MkdirAll = os.MkdirAll
```

Comment:
make these functions, not mutable package vars?

Reply:
The Windows implementation is a function, for Linux we can use the os.MkdirAll implementation.
Is there a better way to accomplish this?

Reply:
a function that delegates to the os function would work and would not be mutable

Reply:
Updated - PTAL